### ISSUE TITLE
Aqara H1 Hub

### ISSUE THREAD

**Issue by cnwangfeng** [niceboygithub/AqaraGateway#28]
Hi Niceboy, 

I am curious about how to setup Aqara H1 Hub because H1 Hub cannot be supported by the Xiaomi Mijia. So, I don't know how to retreive its TOKEN? 

Thanks.

**Comment by rezmus** [comment]
reset wifi, connect to ap and use pocket sender.

**Comment by cnwangfeng** [comment]
@rezmus , thanks a lot for the comment. I can make it connecting to the AP. But I don't know what the "pocket sender"? Do you mean the mobile phone with some APPs? Thanks.

**Comment by rezmus** [comment]
sorry typo i meant pAcket sender. check method 5a/5b from this github

https://github.com/Maxmudjon/com.xiaomi-miio/blob/master/docs/obtain_token.md

when you connect to ap and get ip from dhcp check gw ip (h1 ip) and send this hex to udp 54321 of gw ip.

21310020ffffffffffffffffffffffffffffffffffffffffffffffffffffffff

in response you will get 16 bytes of token. after you have token you can send magic cmd via miio and get telnet access, then make pernament root via post_init.sh or custom firmware. keep in mind that you only have about 10 minutes before ap is closed and you need to restart hub (new token on each reboot).

**Comment by cnwangfeng** [comment]
Hi rezmus, 

I've tried lots of times but still failed. 

1. I reset the H1 and connected my laptop to the WIFI provided by the H1. 
2. I send the UDP datagram to H1's IP address but no response. 
3. I reconfigure the H1 to Aqarahome and send UDP datagram to the specified IP. No response either. 
4. My H1's version: 3.2.7.

**Comment by rezmus** [comment]
it should respond to 54321 udp if miio is running. looking at the firmware it seems like mijia integration is there, but maybe something is broken. make sure it's in mijia mode, delete mi home and install this special version of mi home app where h1 hub was added. if it fails to pair i guess there is no solderless way for this hub.

https://ufile.io/grkwntsb

**Comment by cnwangfeng** [comment]
That's the reason I felt confused. As I know, H1 cannot be supported by Mijia. So I have no way to regsiter the H1 into my Mijia App.

**Comment by rezmus** [comment]
check app i sent you, h1 is added there. some devices have mijia integration and they are not in app.

![Screenshot_2021-07-26-13-09-12-842_com xiaomi smarthome](https://user-images.githubusercontent.com/49418395/126993024-b3866781-f2dd-4df0-81b2-40153aed3c70.png)

**Comment by cnwangfeng** [comment]
Hi @rezmus ,

Thanks a lot. I've been waiting for a few days. There are still no mails in my account. May I ask you sent the mail to my Gmail account?  I just updated my Mijia App. But it's too pity that the H1 can't be found in the App.

Cheers,

**Comment by cnwangfeng** [comment]
Hi @rezmus, 

Are you using Mijia App? I upgrade my Mijia App. H1 still can't be found in the App. Or I am using CN version. Is it different from other countries?

**Comment by niceboygithub** [comment]
@cnwangfeng 
Have you try to add H1 by choosing other Aqara gateway?

**Comment by rezmus** [comment]
delete mi home and install mi home apk from this link.

https://ufile.io/b6zvrhtj

make sure to switch h1 hub to mijia mode before you try to pair it.

**Comment by cnwangfeng** [comment]
Not Yet. I can have a try. Which type would be better？ M1S? P3?

**Comment by cnwangfeng** [comment]
@rezmus

I tried many times today. 
1. I will press the button 10 times and reset the H1 Hub. However, I can't covert it to Xiaomi Mode. It's always in Aqarahome Mode. 
2. I've tried that App. It cannot connect to H1 Hub even I connect to the WIFI of the H1 Hub. 

So, I guess that the new version of H1 Hub (CN Version) has removed all the programs that support Mijia App and MIIO.  It's dedicated to the AQARA HOME.
